src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,835 @@ static DamageTable D_80AD2550 = {
+            break;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            break;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+    if (!(ENOSN_GET_3(THIS))) {
+        if (sp34 != 0) {
+            this->actor.flags |= ACTOR_FLAG_1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = NULL;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, &this->unk_1D8, &this->unk_1DE, this->actor.focus.pos);
+}
+
+s32 EnOsn_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+
+    if (this->unk_1F0 && limbIndex == 11) {
+        Matrix_InsertXRotation_s(this->unk_1D8.y, MTXMODE_APPLY);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 10) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+    Vec3f sp30 = D_80AD2574;
+    Vec3s sp28 = D_80AD2580;
+
+    if (limbIndex == 0xB) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (((this->unk_1EC == 17) || (this->unk_1EC == 21) || (this->unk_1EC == 22)) && (limbIndex == 6)) {
+        Matrix_StatePush();
+        Matrix_InsertTranslation(-400.0f, 1100.0f, -200.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(sp28.x, MTXMODE_APPLY);
+        Matrix_RotateY(sp28.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(sp28.z, MTXMODE_APPLY);
+
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList((*gfx)++, &object_osn_DL_0192A0);
+        Matrix_StatePop();
+    }
+}
+
+void EnOsn_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GraphicsContext* temp_s0;
```

Comment:
```suggestion
    s32 pad;
```
Doesn't look to be used, so should just be normal padding.